inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/phase/ReadPhaseMetricData.java

```diff
@@ -0,0 +1,105 @@
+ *   distributed with this work for additional information
+ *   regarding copyright ownership.  The ASF licenses this file
+ *   to you under the Apache License, Version 2.0 (the
+ *   "License"); you may not use this file except in compliance
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.inlong.sort.base.metric.phase;
+
+import static org.apache.inlong.sort.base.Constants.READ_PHASE_TIMESTAMP;
+
+import java.util.Map;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.inlong.sort.base.metric.MetricData;
+import org.apache.inlong.sort.base.metric.MetricOption;
+import org.apache.inlong.sort.base.metric.ThreadSafeCounter;
+
+/**
+ * A collection class for handling read phase metric data
+ */
+public class ReadPhaseMetricData implements MetricData {
+
+    private final MetricGroup metricGroup;
+    private Counter readPhase;
+    private final Map<String, String> labels;
+
+    public ReadPhaseMetricData(MetricOption option, MetricGroup metricGroup) {
+        this.metricGroup = metricGroup;
+        this.labels = option.getLabels();
+        ThreadSafeCounter readPhaseCounter = new ThreadSafeCounter();
+        readPhaseCounter.inc(option.getReadPhase());
+        registerMetricsForReadPhase(readPhaseCounter);
+    }
+
+    /**
+     * User can use custom counter that extends from {@link Counter}
+     * groupId and streamId and nodeId are label value, user can use it filter metric data when use metric reporter
+     * prometheus
+     */
+    private void registerMetricsForReadPhase(Counter counter) {
+        readPhase = registerCounter(READ_PHASE_TIMESTAMP, counter);
+    }
+
+    /**
+     * Get metric group
+     *
+     * @return The MetricGroup
+     */
+    @Override
+    public MetricGroup getMetricGroup() {
+        return metricGroup;
+    }
+
+    /**
+     * Get labels
+     *
+     * @return The labels defined in inlong
+     */
+    @Override
+    public Map<String, String> getLabels() {
+        return labels;
+    }
+
+    /**
+     * get read phase counter
+     *
+     * @return the read phase counter
+     */
+    public Counter getReadPhase() {
+        return readPhase;
+    }
+
+    /**
+     * output read phase metric,just keep the time to change that moment
+     */
+    public void outputMetrics() {
+        if (readPhase != null) {
+            long count = readPhase.getCount();
+            if (count == 0) {
+                readPhase.inc(System.currentTimeMillis());
```

Comment:
Why increase currenttime since readphase actually means the data count

Reply:
> Why increase currenttime since readphase actually means the data count

This indicator only records the moment when the reading phase changes, so that it is convenient to know when to enter the incremental phase when the indicator is restored.

Reply:
> OK, I think the name readPhase can't be accurate then, pls use a name that's corresponding to its exact meaning

yeah, i will change this name to more understanding name.